src/hooks/useIsValidExecution.ts

```diff
@@ -4,14 +4,45 @@ import type { EthersError } from '@/utils/ethers-utils'
 
 import useAsync from './useAsync'
 import ContractErrorCodes from '@/services/contracts/ContractErrorCodes'
-import { useSafeSDK } from './coreSDK/safeCoreSDK'
+import { type SafeInfo } from '@safe-global/safe-gateway-typescript-sdk'
+import { createWeb3, useWeb3ReadOnly } from '@/hooks/wallets/web3'
+import { type JsonRpcProvider } from '@ethersproject/providers'
+import { type ConnectedWallet } from '@/services/onboard'
+import { getSpecificGnosisSafeContractInstance } from '@/services/contracts/safeContracts'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import useWallet from '@/hooks/wallets/useWallet'
+import { encodeSignatures } from '@/services/tx/encodeSignatures'
+import useIsSafeOwner from '@/hooks/useIsSafeOwner'
 
 const isContractError = (error: EthersError) => {
   if (!error.reason) return false
 
   return Object.keys(ContractErrorCodes).includes(error.reason)
 }
 
+// Monkey patch the signerProvider to proxy requests to the "readonly" provider if on the wrong chain
+const getPatchedSignerProvider = (
+  wallet: ConnectedWallet,
+  chainId: SafeInfo['chainId'],
+  readOnlyProvider: JsonRpcProvider,
+) => {
+  const signerProvider = createWeb3(wallet.provider)
+
+  if (wallet.chainId !== chainId) {
+    const readOnlyMethods = ['eth_getCode', 'eth_call']
```

Comment:
Can you add a comment why we specifically need to patch these method as it is not immediately clear.

```suggestion
    const READ_ONLY_METHODS = ['eth_getCode', 'eth_call']
```